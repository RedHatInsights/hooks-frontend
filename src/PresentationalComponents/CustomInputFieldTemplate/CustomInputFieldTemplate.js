import React from 'react';
import PropTypes from 'prop-types';
import { TextInput, FormGroup } from '@patternfly/react-core';

export class CustomInputFieldTemplate extends React.Component {
    constructor(props) {
        super(props);
        this.state = { value: props.formData ? props.formData : '' };
    }

    state = {
        value: ''
    };

    handleTextInputChange = value => {
        this.props.onChange(value);
        this.setState({ value });
    };

    render() {
        const { value } = this.state;

        return <FormGroup
            label={ this.props.schema.title }
            isRequired={ this.props.required }
            fieldId={ `custom-input-${ this.props.name }` }>
            <TextInput value={ value } aria-label={ this.props.schema.title }
                type="text" onChange={ this.handleTextInputChange } />
        </FormGroup>;
    }
}

CustomInputFieldTemplate.propTypes = {
    formData: PropTypes.string,
    onChange: PropTypes.func,
    schema: PropTypes.object,
    name: PropTypes.string,
    required: PropTypes.bool
};

export default CustomInputFieldTemplate;
