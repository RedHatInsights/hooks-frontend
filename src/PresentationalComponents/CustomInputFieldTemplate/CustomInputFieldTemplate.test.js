import toJson from 'enzyme-to-json';

import { CustomInputFieldTemplate } from './CustomInputFieldTemplate';

describe('CustomInputFieldTemplate', () => {
    const testProps = {
        formData: 'Test Form Data',
        schema: {
            title: 'Test Input Field'
        },
        uiSchema: {
            'ui:placeholder': 'Placeholder text'
        },
        errorSchema: {},
        onChange: jest.fn(),
        name: 'test-input-field',
        required: true
    };

    it('expect to render', () => {
        const wrapper = shallow(
            <CustomInputFieldTemplate { ...testProps } />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('expect to call onChange handler', () => {
        const wrapper = shallow(
            <CustomInputFieldTemplate { ...testProps } />
        );
        const testString = 'Test Input';
        wrapper.find('TextInput').simulate('change', testString);
        expect(testProps.onChange).toHaveBeenCalledWith(testString);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
