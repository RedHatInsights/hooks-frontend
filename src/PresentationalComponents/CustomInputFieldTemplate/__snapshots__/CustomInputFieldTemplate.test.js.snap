// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInputFieldTemplate expect to render 1`] = `
<FormGroup
  className=""
  fieldId="custom-input-test-input-field"
  isInline={false}
  isRequired={true}
  isValid={true}
  label="Test Input Field"
>
  <TextInput
    aria-label="Test Input Field"
    className=""
    isDisabled={false}
    isReadOnly={false}
    isRequired={false}
    isValid={true}
    onChange={[Function]}
    placeholder="Placeholder text"
    type="text"
    value="Test Form Data"
  />
</FormGroup>
`;
